src/libraries/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -954,7 +955,17 @@ internal bool VerifyRemoteCertificate(RemoteCertificateValidationCallback? remot
 
             try
             {
-                _remoteCertificate = CertificateValidationPal.GetRemoteCertificate(_securityContext, out remoteCertificateStore);
+                X509Certificate2? certificate = CertificateValidationPal.GetRemoteCertificate(_securityContext, out remoteCertificateStore);
+
+                if (_remoteCertificate != null && certificate != null &&
+                    Enumerable.SequenceEqual(certificate.RawData, _remoteCertificate.RawData))
```

Comment:
Slightly better to do `Span<byte>.SequenceEqual` instead of using LINQ, I think

Reply:
> Slightly better to do Span<byte>.SequenceEqual instead of using LINQ, I think

Much more than slightly ;-)

Reply:
missed existence of `SequenceEqual` as search for `SequenceEquals` did not show it ;(
updated.